semantic_release/history/__init__.py

```diff
@@ -220,25 +237,36 @@ def get_current_version_by_config_file(omit_pattern=None) -> str:
     return version
 
 
-def get_current_version(prerelease_version: bool = False) -> str:
+def get_current_version() -> str:
     """
     Get current version from tag or version variable, depending on configuration.
-    This will not return prerelease versions.
+    This can be either a release or prerelease version
 
     :return: A string with the current version number
     """
-    omit_pattern = None if prerelease_version else get_prerelease_pattern()
     if config.get("version_source") == "tag":
-        return get_current_version_by_tag(omit_pattern)
-    current_version = get_current_version_by_config_file(omit_pattern)
-    if omit_pattern and omit_pattern in current_version:
-        return get_previous_version(current_version)
-    return current_version
+        return get_current_version_by_tag()
+    else:
+        return get_current_version_by_config_file()
+
+
+def get_current_release_version() -> str:
+    """
+    Get current release version from tag or commit message (no going back in config file),
+    depending on configuration.
+    This will return the current release version (NOT prerelease), instead of just the current version
+
+    :return: A string with the current version number
+    """
+    if config.get("version_source") == "tag":
+        return get_current_release_version_by_tag()
+    else:
+        return get_current_release_version_by_commits()
```

Comment:
`get_current_release_version` should give us the last _not-prerelease_ version.
As the config file does not give us any history, we have to analyse commits here instead of just looking into the config file.